spirv_common.hpp

```diff
@@ -296,6 +296,8 @@ struct SPIRType : IVariant
 		bool depth;
 		bool arrayed;
 		bool ms;
+		bool is_read = false;
```

Comment:
I have a similar situation with samplers in GLSL and HLSL, where I have the distinction between samplerShadow and just plain sampler, but it does not exist in the type system. In that case, I do some hackery to pretend it's part of the type system. In variable_decl I know the variable ID, and I can based on flags modify the type in-place, declare it, and then change it back once the type_to_glsl() has been called. This however, is a bit ugly. The main difference is that at least the shadow information does not leak into other variables.

I think we should try to optionally pass down the variable ID to these functions going forward so that we can deal with this kind of information without hacks and tricks.

Would it be okay for now that all textures are declared as read/write, the is_read/is_write is removed, and then I can have a look at how to refactor the top-level interfaces to deal with this in GLSL/MSL/HLSL?